python_modules/dagster/dagster/_core/definitions/asset_layer.py

```diff
@@ -794,21 +807,25 @@ def build_asset_selection_job(
         build_source_asset_observation_job,
     )
 
+    included_assets = []
```

Comment:
It's a little more verbose, but I think it would make this code more bug-proof to explicitly setting each of these in each of the branches below, instead of setting them up here and overwriting them below.

Reply:
I think the most bulletproof thing here is to make a NamedTuple with these properties.

Reply:
Then you can also extract out functions that return that object as appopriate